### DISCUSSION TITLE
How To Use Python to Control the Drone?

### DISCUSSION THREAD

**Discussion by akshaysunkara** [bitcraze/discussions#1377] [Q&A]
Hey guys, I want to use python to control the crazyflie drone. 

I don't have like the flowdeck or anything but I want to make a simple program that moves the drone based on keyboard presses.

If I'm not able to do it based on keyboard presses, is there another program that I can use that can control the crazyflie drone instead of using the controller.

Thannks a lot

**Comment by gemenerik** [comment]
In theory, it is possible to fly a drone using only the IMU with [our Python library](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/). However, in practice, this is not a good idea unless you use a controller. When relying solely on the IMU, you will experience significant drift, which means you cannot expect any meaningful level of autonomy. While full manual control via a keyboard is theoretically feasible, precise flight control would be very difficult.
